frontend/src/helpers/api.js

```diff
@@ -0,0 +1,35 @@
+import { getEnvVars } from '../environment';
+const { BASE_URL } = getEnvVars();
+
+class Api {
+    client = null;
+
+    async request(route, action, headers, body) {
+        const requestURL = `${BASE_URL}${route}`
+
+        const requestHeaders = {
+            'Content-Type': 'application/json',
+            ...headers
+        };
+
+        try {
+            const response = await fetch(requestURL, {
+                method: action,
+                headers: requestHeaders,
+                body: body
+            });
+
+            const json = await response.json();
+            return json;
+        } catch(error) {
+            return error;
+        }
```

Comment:
This block can be reduced down further and still maintain its legibility.

```suggestion
        return fetch(requestURL, {
            method: action,
            headers: requestHeaders,
            body: body
        }).then(res => res.json()).catch(e => e);
```